### PR TITLE
Adding a build and attach to release workflow (upon release publish).

### PR DESCRIPTION
**Pull request by blckmn**
Simple workflow to attach the assets once a release is published.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by McGiverGim** [comment]
Hi! I only wanted to know how different is your publish workflow, because I don't see too much difference with the other one workflow we have. 
I don't see too much difference between creating an empty publication, and wait to the action to publish the artifacts, or throw the workflow that creates the artifacts and creates an empty publication.
Maybe is there some other step that I'm not taking into account?

**Comment by blckmn** [comment]
The issue is creating artefacts and attaching when a release is in draft is that commits can be added to 'master' (before publishing), invalidating the artefacts. The artefacts need replacing once published.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

File: .github/workflows/build-release.yml
Comment (McGiverGim) [inline .github/workflows/build-release.yml:22]: If we continue with the automatic execution, maybe we can, to be sure that we download what we want, specify the tag to be sure that the checkout is the correct one?

I've seen that this action has a version 3. Maybe is good too start to use the latest version.

File: .github/workflows/build-release.yml
Comment (McGiverGim) [inline .github/workflows/build-release.yml:7]: Only a question, maybe is better to add a manual workflow to attach the build artifacts before the publishing? The difference is not too big, but now, the user subscribed receives the notification of a new version, and the artifacts are not available. Is not too much time, 5 minutes more or less if the build goes well (there is a possibility that it has some kind of problem). 
And the Android version must be generated (and attached I suppose) out of this flow, so the user will see the Android version but nothing more.
Not a big problem, only wanted to notice it.

**Review by McGiverGim — COMMENTED** [review]
Ok, understood. I prefer to have the builds before publishing and not after, but you make the release, so how you prefer and is easy to you. Only two comments.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

File: .github/workflows/build-release.yml
Comment (blckmn) [reply .github/workflows/build-release.yml:22]: Is it not automatically the commit that triggered the run of the job?

**Review by McGiverGim — APPROVED** [review]
Ok. If this is the way you like to work, is ok to me.

Another question: if we launch the "earlier" release flow manually, is this flow released too? If true, then maybe remove the earlier one, it has no sense that the manual one create the artifacts, and this other replace them (or worst, duplicate them).

**Review by haslinghuis — APPROVED** [review]

**Comment by blckmn** [comment]
> Ok. If this is the way you like to work, is ok to me.
> 
> Another question: if we launch the "earlier" release flow manually, is this flow released too? If true, then maybe remove the earlier one, it has no sense that the manual one create the artifacts, and this other replace them (or worst, duplicate them).

Good question. I would hope that it would replace them - or error. I will test that at some point and let you know.

**Pull request merged**
